### PR TITLE
Added password-rules

### PR DESCRIPTION
**Pull request by akshat-max**
### Checklist
- [x] The PR isn't documenting something that would be a common practice among password managers (e.g. minimal length of 6)
- [x] The top-level JSON objects are sorted alphabetically
- [x] There are no [open pull requests](https://github.com/apple/password-manager-resources/pulls) for the same update.

#### for websites-with-shared-credential-backends.json
- [x] There's evidence the domains are related (SSL certificates, DNS entries, valid links between sites, legal documents etc.)
- [x] The new group serves login pages on each of the included domains, and those login page accept accounts from the others. (For example, we don't associate `google.co.il` to `google.com`, because `google.co.il` redirects to `accounts.google.com` for authentication.)

#### for change-password-URLs.json
- [x] There is no Well-Known URL for Changing Passwords (`https://example.com/.well-known/change-password`)
- [x] The URL either makes the experience better or no worse than being directed to just the domain in a non-logged-in state

#### for password-rules.json
- [x] The given rule isn't particularly standard and obvious for password managers
- [x] Generated passwords have been tested from this rule using the [Password Rules Validation Tool](https://developer.apple.com/password-rules/)
- [x] Information has been included about the website's requirements (eg. screenshots, error messages, steps during experimentation, etc.)

### CODE REVIEW

File: quirks/password-rules.json [hunk 1]
```diff
@@ -255,6 +258,9 @@
     "ubisoft.com": {
         "password-rules": "minlength: 8; maxlength: 16; required: lower; required: upper; required: digit; required: [-]; required: [!@#$%^&*()+];"
     },
+    "Udacity.com": {
```
Comment (rmondello) [inline quirks/password-rules.json:261]: Can you lower-case this U?

Comment (akshat-max) [reply quirks/password-rules.json]: Done required change

**Comment by rmondello** [comment]
Hi @akshat-max! I’d love to merge this. Can you verify first that you agree to the project’s Developer Certificate of Origin? https://github.com/apple/password-manager-resources/blob/main/DEVELOPER_CERTIFICATE_OF_ORIGIN.md

**Comment by rmondello** [comment]
Closing due to inactivity. Feel free to re-open any time!

**Pull request closed without merging**